src/ProductConstructionService/ProductConstructionService.DependencyFlow/SubscriptionTriggerer.cs

```diff
@@ -149,10 +150,19 @@ await AddDependencyFlowEventAsync(
             })
             .ToList();
 
-        await _redisMutex.EnterWhenAvailable(
-            pullRequestUpdater.Id.ToString(),
-            async () =>
+
+        IAsyncDisposable? @lock;
```

Comment:
do we need this lock here? We won't the lock in the `WorkItemScope` be enough?

Reply:
Ah the IDs will be different because we don't have to batched `UpdaterId` then, right?

Reply:
I'll have a look

Reply:
yeah I think that's the case. In the `SubscriptionTriggererWorkItemProcessor` we only have the SubId. For non batched subs, the PullRequestUpdaterId is the same as the SubId, but not for batched, so we need to check the lock when we get it

Reply:
Yeah, it's what you say, we first need to get the ID